waverecoverygui.py

```diff
@@ -0,0 +1,340 @@
+import sys
+from argparse import Namespace
+from PyQt5.QtWidgets import (
+    QApplication,
+    QMainWindow,
+    QWidget,
+    QLabel,
+    QPushButton,
+    QTextEdit,
+    QFileDialog,
+    QFormLayout,
+    QComboBox,
+    QCheckBox,
+    QSpinBox,
+    QMessageBox,
+    QProgressBar,
+    QHBoxLayout,
+)
+from PyQt5.QtCore import QThread, pyqtSignal
+import io
+from contextlib import redirect_stdout
+from waveheaderprocessor import WaveHeaderProcessor
+from PyQt5.QtGui import QTextCursor, QTextCharFormat, QColor, QIntValidator
+
+class WorkerThread(QThread):
+    update_progress = pyqtSignal(int)
+    update_log = pyqtSignal(str, bool)
+
+    def __init__(self, args):
+        super().__init__()
+        self.args = args
+
+    def run(self):
+        try:
+            with io.StringIO() as buf, redirect_stdout(buf):
+                processor = WaveHeaderProcessor()
+                total_files = processor.repair_audio_file_headers(
+                    self.args.source_path,
+                    self.args.destination_path,
+                    self.args.sample_rate,
+                    self.args.bits_per_sample,
+                    self.args.channels,
+                    self.args.verbose,
+                    self.args.force,
+                    self.args.application,
+                    self.args.offset,
+                    self.args.end_offset,
+                )
+                output = buf.getvalue()
+
+            self.update_progress.emit(total_files)
+            self.update_log.emit(output, False)
+
+        except Exception as e:
+            error_message = f'Error: {str(e)}'
+            self.update_log.emit(error_message, True)
+            QMessageBox.critical(self.args.gui, 'Error', error_message)
+        finally:
+            self.args.gui.restore_button.setDisabled(False)
+
+class WaveRecoveryToolGUI(QMainWindow):
+    def __init__(self):
+        super().__init__()
+
+        # Initialize a list of worker threads
+        self.worker_threads = []
+
+        self.initUI()
+
+    def initUI(self):
+        self.setWindowTitle('Wave Recovery Tool GUI')
+        self.setGeometry(100, 100, 200, 200)

```

Comment:
The window is quite small and the paths are not readable after browsing. I would propose to increase the width of the window a bit.